rust/fastpay/src/client.rs

```diff
@@ -278,101 +277,113 @@ fn deserialize_response(response: &[u8]) -> Option<AccountInfoResponse> {
-            .args_from_usage("
-            --from=<ADDRESS>     'Sending address (must be one of our accounts)'
-            --to=<ADDRESS>       'Recipient address'
-            <amount>             'Amount to transfer'
-            "))
-        .subcommand(SubCommand::with_name("query_balance")
-            .about("Obtain the spendable balance")
-            .args_from_usage("
-            <address>            'Address of the account'
-            "))
-        .subcommand(SubCommand::with_name("benchmark")
-            .about("Send one transfer per account in bulk mode")
-            .arg(
-                Arg::with_name("max_in_flight")
-                    .long("max_in_flight")
-                    .help("Maximum number of requests in flight")
-                    .default_value("200"),
-            )
-            .arg(
-                Arg::with_name("max_orders")
-                    .long("max_orders")
-                    .help("Use a subset of the accounts to generate N transfers")
-                    .default_value(""),
-            )
-            .arg(
-                Arg::with_name("server_configs")
-                    .long("server_configs")
-                    .help("Use server configuration files to generate certificates (instead of aggregating received votes).")
-                    .min_values(1),
-            ))
-        .subcommand(SubCommand::with_name("create_accounts")
-            .about("Create new user accounts and print the public keys")
-            .arg(
-                Arg::with_name("initial_funding")
-                    .long("initial_funding")
-                    .help("known initial balance of the account")
-                    .default_value("0"),
-            )
-            .args_from_usage("
-            <num>                'Number of additional accounts to create'
-            "))
-        .get_matches();
-
-    let send_timeout =
-        Duration::from_micros(matches.value_of("send_timeout").unwrap().parse().unwrap());
-    let recv_timeout =
-        Duration::from_micros(matches.value_of("recv_timeout").unwrap().parse().unwrap());
-    let accounts_config_path = matches.value_of("accounts").unwrap();
-    let committee_config_path = matches.value_of("committee").unwrap();
-    let buffer_size = matches
-        .value_of("buffer_size")
-        .unwrap()
-        .parse::<usize>()
-        .unwrap();
+    let matches = ClientOpt::from_args();
+
+    let send_timeout = Duration::from_micros(matches.send_timeout);
+    let recv_timeout = Duration::from_micros(matches.recv_timeout);
+    let accounts_config_path = &matches.accounts;
+    let committee_config_path = &matches.committee;
+    let buffer_size = matches.buffer_size.parse::<usize>().unwrap();
 
     let mut accounts_config =
-        AccountsConfig::read_or_create(accounts_config_path).expect("Unable to read user accounts");
+        AccountsConfig::read_or_create(&accounts_config_path).expect("Unable to read user accounts");
```

Comment:
This `&` is surprising because `&matches.accounts` is already a reference.

Reply:
good point. it works without the & also so i'll remove it